Azure-Samples/azure-search-openai-demo
# Get PDFs to jump to their pages again

## Purpose

Fixes #1245

This PR has a few fixes, but the biggest one is a fix for regression introduced by #1160 which replaced citation URLs with blob URLs. My fix remembers the hash and adds it to the blob URL after, which seems to work correctly in Edge. I couldn't find any documentation as to whether blob URLs formally support hash fragments, but they seem to!

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Click on citations in answers that have #page= hash, see PDF viewer jump to that page
* Switch to GPT4V env, click on citations with no hash, see them open fine
